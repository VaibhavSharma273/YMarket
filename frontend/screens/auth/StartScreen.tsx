import { StyleSheet } from 'react-native';

import EditScreenInfo from '../../components/EditScreenInfo';
import { Text, View } from '../../components/Themed';
import { TouchableOpacity } from 'react-native'
// import { RootStackParamList, RootTabScreenProps } from '../types';
// import { StackScreenProps } from '@react-navigation/stack';

import Logo from "../../components/Logo"

export default function StartScreen({ navigation }: any) {
  return (
    <View style={styles.container}>
        <Logo/>
      <View style={styles.separator} />
      <TouchableOpacity 
        style={styles.button}
        onPress={() => navigation.navigate('LoginScreen')}>
        <Text style={styles.title}>Login</Text>
        </TouchableOpacity>
        <View style={styles.separator}/>
      <TouchableOpacity 
        style={styles.button}
        onPress={() => navigation.navigate('RegisterScreen')}>
        <Text style={styles.title}>Register</Text>
        </TouchableOpacity>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  title: {
    fontSize: 20,
    fontWeight: 'bold',
    color: 'white',
    fontFamily: 'Arial',
  },
  separator: {
    marginVertical: 5,
    height: 3,
    width: '50%',
  },
  header: {
    fontSize: 21,
    color: '#0f4d92',
    fontWeight: 'bold',
    paddingVertical: 12,
  },
  button: {
    alignItems: "center",
    backgroundColor: "#0f4d92",
    padding: 10,
    width: 170,
  }
});
