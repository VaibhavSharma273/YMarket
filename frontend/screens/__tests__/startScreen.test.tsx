import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import StartScreen from '../auth/StartScreen';
import { TouchableOpacity } from 'react-native';

describe('Start Screen component', () => {
    it(`renders correctly`, () => {
        const component = shallow(<StartScreen />);
        expect(component.length).toBe(1);
    });

    it('matches the snapshot', () => {
        // const component = shallow(<StartScreen/>);
        // expect(component.getElements()).toMatchSnapshot();
        const tree = renderer.create(<StartScreen />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('Test Touchable Opacity Component', () => {
    it('Test click event', () => {
        const mockClick = jest.fn();
        const touchableOpacity = shallow(<TouchableOpacity onPress={mockClick}>Login</TouchableOpacity>);
        touchableOpacity.find('TouchableOpacity').first().props().onPress();
        expect(mockClick).toHaveBeenCalledTimes(1);
    });
});
