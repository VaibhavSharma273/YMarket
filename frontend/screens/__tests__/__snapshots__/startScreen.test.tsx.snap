// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Start Screen component matches the snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 150,
        "marginBottom": 10,
        "width": 170,
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "height": 3,
          "marginVertical": 5,
          "width": "50%",
        },
      ]
    }
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#0f4d92",
        "opacity": 1,
        "padding": 10,
        "width": 170,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
          },
          Object {
            "color": "white",
            "fontFamily": "Arial",
            "fontSize": 20,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Login
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "height": 3,
          "marginVertical": 5,
          "width": "50%",
        },
      ]
    }
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#0f4d92",
        "opacity": 1,
        "padding": 10,
        "width": 170,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
          },
          Object {
            "color": "white",
            "fontFamily": "Arial",
            "fontSize": 20,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Register
    </Text>
  </View>
</View>
`;
